eclipse-collections-code-generator/src/main/resources/test/list/mutable/boxedMutableArrayListTest.stg

```diff
@@ -0,0 +1,165 @@
+    }
+
+    @Test
+    public void lastIndexOf()
+    {
+        BoxedMutable<name>List list = this.classUnderTest();
+        list.add(<(literal.(type))("1")>);
+        Assert.assertEquals(3, list.lastIndexOf(<(literal.(type))("1")>));
+        Assert.assertEquals(1, list.lastIndexOf(<(literal.(type))("2")>));
+        Assert.assertEquals(-1, list.indexOf(<(literal.(type))("5")>));
+    }
+
+    @Test
+    public void set()
+    {
+        BoxedMutable<name>List list = this.classUnderTest();
+        Assert.assertEquals(<(literal.(type))("2")>, (<type>) list.set(1, <(literal.(type))("5")>)<wideDelta.(type)>);
+        Assert.assertEquals(new BoxedMutable<name>List(new <name>ArrayList(<["1", "5", "3"]:(literal.(type))(); separator=", ">)), list);
+
+        Assert.assertEquals(<(literal.(type))("5")>, (<type>) list.set(1, <(literal.(type))("7")>)<wideDelta.(type)>);
+        Assert.assertEquals(new BoxedMutable<name>List(new <name>ArrayList(<["1", "7", "3"]:(literal.(type))(); separator=", ">)), list);
+    }
+
+    @Test
+    public void mutationOfOriginalList()
+    {
+        <name>ArrayList originalList = new <name>ArrayList(<["1", "2", "3"]:(literal.(type))(); separator=", ">);
+        BoxedMutable<name>List list = new BoxedMutable<name>List(originalList);
+        Assert.assertEquals(list, Lists.mutable.of(<["1", "2", "3"]:(literal.(type))(); separator=", ">));
+
+        originalList.add(<(literal.(type))("4")>);
+        Assert.assertEquals(list, Lists.mutable.of(<["1", "2", "3", "4"]:(literal.(type))(); separator=", ">));
+
+        originalList.remove(<(literal.(type))("2")>);
+        Assert.assertEquals(list, Lists.mutable.of(<["1", "3", "4"]:(literal.(type))(); separator=", ">));
+
+        originalList.addAllAtIndex(1, <["7", "8", "9"]:(literal.(type))(); separator=", ">);
+        Assert.assertEquals(list, Lists.mutable.of(<["1", "7", "8", "9", "3", "4"]:(literal.(type))(); separator=", ">));
+
+        originalList.clear();
+        Verify.assertEmpty(list);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void setIndexOutOfBounds()
+    {
+        this.classUnderTest().set(5, <(literal.(type))("5")>);
+    }
+
+    @Test
+    public void get()
+    {
+        BoxedMutable<name>List list = this.classUnderTest();
+        Assert.assertEquals(<(literal.(type))("1")>, (<type>) list.get(0)<wideDelta.(type)>);
+        Assert.assertEquals(<(literal.(type))("2")>, (<type>) list.get(1)<wideDelta.(type)>);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void getIndexOutOfBounds()
+    {
+        this.classUnderTest().get(5);
+    }
+
+    @Test
+    public void addAll()
+    {
+        BoxedMutable<name>List list = this.classUnderTest();
+        list.addAll(1, Lists.mutable.of(<(literal.(type))("7")>, <(literal.(type))("9")>));
+        Assert.assertEquals(new BoxedMutable<name>List(new <name>ArrayList(<["1", "7", "9", "2", "3"]:(literal.(type))(); separator=", ">)), list);
```

Comment:
I think this should be a JDK `ArrayList` or EC `FastList` of `Boolean` to make sure the equals contract works correctly.

Reply:
Fixed. Thanks.